### PR TITLE
Better use of Vault api.

### PR DESCRIPTION
**Pull request by Emibergo02**
Withdraw already checks if the player has the necessary amount.
Updated RedisEconomy API to 4.0 and added transaction reasons

### CODE REVIEW

File: common/src/main/java/net/william278/husktowns/manager/TownsManager.java [hunk 1]
```diff
@@ -689,7 +688,7 @@ public void withdrawMoney(@NotNull OnlineUser user, @NotNull BigDecimal amount)
 
             final Town town = member.town();
             final BigDecimal withdrawal = amount.min(town.getMoney());
-            economy.giveMoney(user, withdrawal);
+            economy.giveMoney(user, withdrawal, "Town "+town.getId()+":"+town.getName()+" withdrawal");
```
Comment (WiIIiam278) [inline common/src/main/java/net/william278/husktowns/manager/TownsManager.java:691]: style: spaces between plus symbols in string concatenation `+`

File: common/src/main/java/net/william278/husktowns/manager/TownsManager.java [hunk 2]
```diff
@@ -656,13 +656,12 @@ public void depositMoney(@NotNull OnlineUser user, @NotNull BigDecimal amount) {
             }
 
             final Town town = member.town();
-            if (!economy.hasMoney(user, amount)) {
+            if(!economy.takeMoney(user, amount, "Town "+town.getId()+":"+town.getName()+" deposit")){
```
Comment (WiIIiam278) [inline common/src/main/java/net/william278/husktowns/manager/TownsManager.java:659]: style: spaces between plus symbols in string concatenation (`+`).
Use IntelliJ ALT+ENTER autoformatting if possible.

File: common/src/main/java/net/william278/husktowns/hook/EconomyHook.java [hunk 3]
```diff
@@ -14,9 +15,16 @@ protected EconomyHook(@NotNull HuskTowns plugin, @NotNull String name) {
 
     public abstract boolean hasMoney(@NotNull OnlineUser user, @NotNull BigDecimal amount);
```
Comment (WiIIiam278) [inline common/src/main/java/net/william278/husktowns/hook/EconomyHook.java:16]: Do we still need `hasMoney`? If not, get rid of it. I assume we probably still do if it's used elsewhere.

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]
A bit of style and a question :)

File: common/src/main/java/net/william278/husktowns/manager/TownsManager.java (hunk 1 above)
Comment (Emibergo02) [reply common/src/main/java/net/william278/husktowns/manager/TownsManager.java]: i forgot Ctrl+Alt+L

File: common/src/main/java/net/william278/husktowns/hook/EconomyHook.java (hunk 3 above)
Comment (Emibergo02) [reply common/src/main/java/net/william278/husktowns/hook/EconomyHook.java]: nope. i also removed an unused getBalance

**Review by WiIIiam278 — APPROVED** [review]

**Pull request merged**